docs: avoid concurrency issues when generating images in parallel

tarikgraba opened the pull request:
This should resolve issue #4631 when building the doc images in parallel.

I have removed the `tidy` target and I just delete log and aux file after building each pdf.

KrystalDelusion commented on docs/source/_images/Makefile:
```diff
@@ -31,14 +31,13 @@ all_tex: $(TEX_FILES:.tex=.pdf) $(TEX_FILES:.tex=.svg)
 
 %.pdf: %.tex
 	cd $(@D) && $(FAKETIME) pdflatex $(<F) --interaction=nonstopmode
+	rm -f $(*).aux $(*).log
```
I think the rm was there because of how the docs website was being done and is no longer necessary. It would be better to leave this out entirely and just rely on the gitignore instead of removing the log files.

KrystalDelusion reviewed (commented):
Thanks for doing this!  I always run make docs without a -j flag and have been meaning to look into it for ages but kept forgetting.

tarikgraba commented:
I have reverted my previous commit and just removed the `tidy` target from the main target as I understand it is not needed any more (aux/log are ignored in `docs/.gitignore`)
